lib/SPIRV/SPIRVToOCL.cpp

```diff
@@ -642,6 +642,12 @@ void SPIRVToOCLBase::visitCallGenericCastToPtrExplicitBuiltIn(CallInst *CI,
   default:
     llvm_unreachable("Invalid address space");
   }
+
+  AttributeList Attrs = CI->getCalledFunction()->getAttributes();
+  SmallVector<AttributeSet, 1> ArgAttrs = {Attrs.getParamAttrs(0)};
+  AttributeList NewAttrs = AttributeList::get(*Ctx, Attrs.getFnAttrs(),
+                                              Attrs.getRetAttrs(), ArgAttrs);
```

Comment:
Why not use [`AttributeList::removeParamAttributes`](https://llvm.org/doxygen/classllvm_1_1AttributeList.html#a14fe429b8e7e3af4cab47a826ae6918f)?

Reply:
Actually https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/lib/SPIRV/SPIRVToOCL.cpp#L648 will call [AttributeList::removeParamAttributes](https://llvm.org/doxygen/classllvm_1_1AttributeList.html#a14fe429b8e7e3af4cab47a826ae6918f), but it don't work. We still have 2 function parameter attribute although the 1th argument is removed.

Reply:
I feel like there is a bug in `removeArg` implementation, because it only modifies internal field `Attrs` of `BuiltinCallMutator` class instance, but I don't see it applying new attributes for the resulting instruction anywhere.

If we already try to handle attributes in that helper, I would prefer us to fix that helper instead of adding such quickfixes everywhere: I suppose that this is not the only place where we leave extra attributes after removing call arguments.

Reply:
@AlexeySachkov, New attributes is applied on https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/lib/SPIRV/SPIRVBuiltinHelper.cpp#L102,L104, bug is from https://github.com/llvm/llvm-project/blob/main/llvm/lib/IR/Attributes.cpp#L1422 which cause after call removeParamAttributes function, Attrs.getNumAttrSets() still unchanged.  Maybe I can submit patch to LLVM, SPIRV don't need any change.
+@jcranmer-intel

Reply:
If there is a bug in the upstream LLVM, we should definitely report it.

If you have a strong case with this particular instruction: I think we can merge this as a temporary workaround, but we should left a comment referencing upstream issue and a test for this case. 

However, my preference would be to fix this in the upstream first, because we are likely have more instructions for which we have the same bug with extra attributes: ideally, we should fix them all at once and not to insert a workaround for each.

Reply:
Filed an issue in LLVM https://github.com/llvm/llvm-project/issues/59746

Reply:
@AlexeySachkov, I don't know if the AttributeList::removeParamAttributes has bug or not so I filed an issue in LLVM. And the patch removes AttributeList::removeParamAttributes call in SPIRVBuiltinHelper.cpp so we don't need add corner case for every instruction that need remove arg. Thanks very much for your review.

Reply:
I'm OK with the fix with TODO placed in the code and in the translator's issues.

Reply:
done.